app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -323,6 +327,30 @@ public void onConfigurationChanged(Configuration newConfig) {
         if (drawerToggle != null) { // Tablet layout does not have a drawer
             drawerToggle.onConfigurationChanged(newConfig);
         }
+        setNavDrawerSize();
+    }
+
+    private void setNavDrawerSize() {
+        if (isTablet()) {
+            return;
+        }
+        float screenPercent = getResources().getInteger(R.integer.nav_drawer_screen_size_percent) * 0.01f;
+        int width = (int) (getScreenWidth() * screenPercent);
+        int maxWidth = (int) getResources().getDimension(R.dimen.nav_drawer_max_screen_size);
+
+        navDrawer.getLayoutParams().width = Math.min(width, maxWidth);
+    }
+
+    private int getScreenWidth() {
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
+        return displayMetrics.widthPixels;
+    }
+
+    private boolean isTablet() {
+        return (getResources().getConfiguration().screenLayout
+                & Configuration.SCREENLAYOUT_SIZE_MASK)
+                >= Configuration.SCREENLAYOUT_SIZE_LARGE;
```

Comment:
It would be more reliable to check if `drawerToggle == null` to detect if we are using the tablet layout.